parsing/builtin_attributes.ml

```diff
@@ -243,34 +351,36 @@ let warning_scope ?ppwarning attrs f =
     Warnings.restore prev;
     raise exn
 
-
-let warn_on_literal_pattern =
+let has_attribute nms attrs =
   List.exists
-    (fun a -> match a.attr_name.txt with
-       | "ocaml.warn_on_literal_pattern"|"warn_on_literal_pattern" -> true
-       | _ -> false
-    )
-
-let explicit_arity =
-  List.exists
-    (fun a -> match a.attr_name.txt with
-       | "ocaml.explicit_arity"|"explicit_arity" -> true
-       | _ -> false
-    )
-
-let immediate =
-  List.exists
-    (fun a -> match a.attr_name.txt with
-       | "ocaml.immediate"|"immediate" -> true
-       | _ -> false
-    )
-
-let immediate64 =
-  List.exists
-    (fun a -> match a.attr_name.txt with
-       | "ocaml.immediate64"|"immediate64" -> true
-       | _ -> false
-    )
+    (fun a ->
+       if List.mem a.attr_name.txt nms
+       then (mark_used a.attr_name; true)
+       else false)
+    attrs
+
+type attr_action = Mark_used_only | Return
+let filter_attributes actions attrs =
```

Comment:
`filter` or `select`? I am also wondering if we should focus on the action on the attribute or the status (`Active | Inactive` ) of the attribute  (with the same semantic).

Reply:
I agree select is clearer and have changed it.